src/test/groovy/io/openliberty/tools/gradle/VerifyFeatureTest.groovy

```diff
@@ -0,0 +1,126 @@
+package io.openliberty.tools.gradle
+
+import static org.junit.Assert.*
+
+import org.junit.BeforeClass
+import org.junit.After
+import org.junit.AfterClass
+import org.junit.FixMethodOrder
+import org.junit.Test
+import org.junit.runners.MethodSorters
+import io.openliberty.tools.common.plugins.util.PluginExecutionException
+import org.gradle.testkit.runner.BuildResult
+
+class VerifyFeatureTest extends AbstractIntegrationTest{
+    static File resourceDir = new File("build/resources/test/prepare-feature-test")
+	static File resourceDir2 = new File("build/resources/test/verify-feature-test")
+    static File buildDir = new File(integTestDir, "/VerifyFeature")
+	static File resourceSimpleBom = new File(resourceDir, "SimpleActivator-bom-1.0.pom")
+	static File resourceSimpleEsa = new File(resourceDir, "SimpleActivatorESA-1.0.esa")
+	static File resourceSimpleAsc = new File(resourceDir, "SimpleActivatorESA-1.0.esa.asc")
+	static File mavenLocalRepo = new File(System.getProperty("user.home")+ "/.m2/repository")
+	static File userTestRepo = new File(mavenLocalRepo, "test/user/test/osgi")
+	static File simpleBom = new File(userTestRepo, "SimpleActivator-bom/1.0/SimpleActivator-bom-1.0.pom")
+	static File simpleEsa = new File(userTestRepo, "SimpleActivatorESA/1.0/SimpleActivatorESA-1.0.esa")
+	static File simpleAsc = new File(userTestRepo, "SimpleActivatorESA/1.0/SimpleActivatorESA-1.0.esa.asc")
+	static File simpleValidKey = new File(buildDir, "src/test/resources/SimpleActivatorValidKey.asc")
+    def featureFile = new File(buildDir, "build/wlp/usr/extension/lib/features/test.user.test.osgi.SimpleActivator.mf")
+    
+
+    @BeforeClass
+    public static void setup() {
+        createDir(buildDir)
+        createTestProject(buildDir, resourceDir2, "build.gradle")
+        copyFile(resourceSimpleBom, simpleBom)
+		copyFile(resourceSimpleEsa, simpleEsa)	
+        copyFile(resourceSimpleAsc, simpleAsc)
+    }
+	
+	@After
+	public void cleanup() {
+		featureFile.delete()
+	}
+
+    
+    @Test
+    public void test_verifyEnforce() {
+        try {
+            System.properties['verify'] = 'enforce'
+            System.properties['keyid'] = ''
+            runTasks(buildDir, 'installFeature')
+
+            assert featureFile.exists() : "SimpleActivator.mf cannot be generated"
+            
+        } catch (Exception e) {
+            throw new AssertionError ("Verify \"enforce\" should pass", e)
+        }
+    }
+
+
+
+    //TODO: Disable for now. 
```

Comment:
Should this be reenabled before merging? If not, why not?

Reply:
It will have to wait until https://github.com/OpenLiberty/open-liberty/pull/26602 is released. I believe customers won't hit the problem unless they run multiple installFeature task in the same build.

Reply:
Ok, then open an issue in our repo to follow up on that please. When do you expect that fix to go in?

Reply:
Yup, I just opened the issue. I don't think the fix made it into 23.0.0.11 so it will be 23.0.0.12.